libs/knowledge-store/ragstack_knowledge_store/knowledge_store.py

```diff
@@ -342,28 +420,127 @@ def _linked_ids(
         results = self._session.execute(self._query_linked_ids, (source_id,))
         return _results_to_ids(results)
 
+    def mmr_traversal(
```

Comment:
atm, we call the method `traversing_retrieve` in #452 .
Should we call it `traversal` for consistency ?

Reply:
I think we should figure out a pattern, but it can wait until we start stabilizing things. I think maybe they should all have retrieve, but in VectorStore they have `similarity_search` which doesn't? So maybe these are instead `traverse` and `max_marginal_relevance_traverse`?

Looking at https://github.com/langchain-ai/langchain/blob/master/libs/core/langchain_core/vectorstores.py#L164, we should also make them available via `search`, and that shows the different names of the specific methods...

Reply:
Right, but the question is still the method names (and to some degree how that maps to names in `search`).

I think that `similarity_search` as a method name is meant as a verb (you do "similarity search") rather than a noun (you perform "a similarity search"). Since they also have `max_marginal_relevance_search`, it makes me wonder:

I think either of these make sense:

* `traverse` and `max_marginal_relevance_traverse` -- the thing we do is "traverse" rather than "search"
* `traversing_search` and `max_marginal_relevance_traversing_search` -- the thing we do is still search, but we add more modifiers. these names feel long...
* maybe `traversal_search` and `max_marginal_relevance_traversal_search` -- using traversal instead of traversing

For `search_type` it seems like they omit the `_search` from the method name (and abbreviate things). So that would suggest:

* method: `traversing_search`, `search_type: 'traversing'`
* method: `mmr_traversing_search` (or `max_marginal_relevance_traversing_search`), `search_type: `mmr_traversing`)

(although `traversal` also seems good as the word).

I don't have strong opinions beyond consistency (probably `_search` suffix makes sense, etc.) -- any thoughts on your side?

Reply:
I like traversal_search